src/filter/DatasetSimilarityFilter.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING, Type
+from src.filter.SampleMetricFilter import SampleMetricFilter
+from src.core.Setupable import SetupMode
+from src.metric.DatasetSimilaritySampleMetric import DatasetSimilaritySampleMetric
+import numpy as np
+import pandas as pd
+
+if TYPE_CHECKING:
+    from src.metric.SampleMetric import SampleMetric
+    from src.metric.SampleMetricManager import SampleMetricManager
+
+DATASET_SIMILARITY_FILTER_NAME = "DatasetSimilarity"
+
+
+class DatasetSimilarityFilter(SampleMetricFilter):
+    """
+    Implementation of an dataset similarity filter.
+
+    This filter measures a similarity score between samples in a population and a dataset.
+    Each sample in the population gets a similarity score based on the "closest"
+    sample in the dataset. The idea is to filter out look-a-likes, i.e., when it's
+    obvious that the generator has produced images which is copied directly from the dataset,
+    those should be removed to mitigate identity leakage.
+    """
+
+    def reg_setup_modes(self) -> dict[str, SetupMode]:
+        return self.sample_metric_reg_setup_modes()
+
+    def apply(
```

Comment:
this function does not match super-method

I suggest adding a kw-arg paramter to the superclass, and to move the three contradicting parameters declared here such that they are extracted in the body from the kw-arg instead. The docstring should be kept the same.

Reply:
Ah, nice catch. Thanks for the suggestion solution, i did it that way!